cli/js/test_util.ts

```diff
@@ -131,22 +142,81 @@ export function testPerm(perms: TestPermissions, fn: Deno.TestFunction): void {
     return;
   }
 
-  Deno.test(fn.name, assertResources(fn));
+  Deno.test(fn.name, sanitizeResources(fn));
+}
+
+interface UnitTestOptions {
+  name?: string;
+  skip?: boolean;
+  perms?: TestPermissions;
+}
+
+export function unitTest(fn: Deno.TestFunction): void;
+export function unitTest(name: string, fn: Deno.TestFunction): void;
+export function unitTest(options: UnitTestOptions, fn: Deno.TestFunction): void;
+export function unitTest(
+  name: string,
+  options: UnitTestOptions,
+  fn: Deno.TestFunction
+): void;
+export function unitTest(
+  optionsOrNameOrFn: UnitTestOptions | string | Deno.TestFunction,
+  maybeFnOrOptions?: UnitTestOptions | Deno.TestFunction,
+  maybeFn?: Deno.TestFunction
+): void {
+  assert(optionsOrNameOrFn, "At least one argument is required");
+
+  let options: UnitTestOptions;
+  let name: string;
+  let fn: Deno.TestFunction;
+
+  if (typeof optionsOrNameOrFn === "string") {
+    name = optionsOrNameOrFn;
+    assert(maybeFnOrOptions, "Missing test function definition");
+    if (typeof maybeFnOrOptions === "function") {
+      fn = maybeFnOrOptions;
+      options = {};
+    } else {
+      options = maybeFnOrOptions;
+      assert(maybeFn, "Missing test function definition");
+      fn = maybeFn;
+    }
+  } else if (typeof optionsOrNameOrFn === "function") {
+    options = {};
+    fn = optionsOrNameOrFn;
+    name = fn.name;
+    assert(name, "Missing test function name");
+  } else {
+    options = optionsOrNameOrFn;
+    assert(maybeFnOrOptions, "Missing test function definition");
+    assert(
+      typeof maybeFnOrOptions === "function",
+      "Second argument should be test function definition"
+    );
+    fn = maybeFnOrOptions as Deno.TestFunction;
+    name = options.name || fn.name;
+    assert(name, "Missing test name");
+  }
+
+  if (options.skip) {
+    return;
+  }
+
+  const normalizedPerms = normalizeTestPermissions(options.perms || {});
+  registerPermCombination(normalizedPerms);
+  if (!permissionsMatch(processPerms, normalizedPerms)) {
+    return;
+  }
+
+  const testDefinition: Deno.TestDefinition = {
+    name,
+    fn: sanitizeResources(fn)
+  };
```

Comment:
Note to self, if `Deno.TestDefinition` was to support `skip` option, then permissions check could be implemented as `skip` assertion - it could show up in report as skipped.

Also brainstorm: better permissions discovery and running tests in different processes (maybe JSON output - trivial to parse)

Reply:
I don't know about that. Skipped tests should show up in the report as
```
SKIPPED     netCloseReadSuccess
```
so that would be a lot of clutter.

Reply:
Good point! I got a [branch](#4235) which implements communication with subprocess using JSON over stdout 😄 it implements what we discussed in #4166. Using it we can filter and format however we please.